### ISSUE TITLE
Sass image URLs always point to dev bucket

### ISSUE THREAD

**Issue by brlodi** [UrbanCCD-UChicago/plenario2#387]
The new landing page and general design upgrades from #381 use responsive image assets stored in an S3 bucket because they take several minutes to generate and that would bog down the local dev builds. Phoenix is set up to rewrite the S3 URLs in local development, `dev.plenar.io` (staging), and `plenar.io` (production) to the corresponding bucket based on where the Phoenix server is running.

However, do to an oversight nothing is handling the URLs included in our Sass files, since Phoenix is not directly processing them (they're built by Webpack). They're statically pointed at the `plenario2-assets-dev` bucket.

Ideally everything should be configured in one place, so I don't want to add a new Webpack stanza mirroring the Phoenix config just to handle URLs in Sass (plus it would require a different mechanism than Phoenix is using).

This issue is soliciting ideas for how to resolve the Sass problem specifically, or alternatively how to redesign the process of S3 deployment to accommodate both HTML/CSS and perhaps automation of the currently manual process.

**Comment by vforgione** [comment]
I think we could just add this as config value. We would need to add a `stage.exs` and update the build process (which is something I want to do anyway). 

In the case of true VueJS things: it's still going to be digested by Phoenix. I'm assuming the call to `mix phx.digest` and `mix phx.server` can inject the environment name (`:dev`, `:stage`, `:prod`, `:test`) into the build environment Webpack uses.

**Comment by brlodi** [comment]
I'm sure we can write the appropriate `mix` scriptlets to achieve that, though by Node convention there are only two environments—`development` and `production`—so we can't just directly inject the environment variable

**Comment by vforgione** [comment]
Environment name in so far as the application's environment so we can build the paths:

```javascript
const imgUrlBase = `https://s3.aws.com/plenario-image-cache-${process.env.MIX_ENV}`
```

_Disclaimer: I'm sure my code snippet is wrong, but you get the gist of it..._

**Comment by brlodi** [comment]
Phoenix doesn't build anything for Webpack, it just literally calls `npm run watch` when run in development (for production you have to manually call `npm run build && mix phx.digest`). We'll need a `webpack.stage.js`, configure the JS environment variable appropriately in there (as well as the equivalents in `webpack.prod.js` and `webpack.dev.js`), and then to add a new script in `package.json` and update the deployment script to call it.

**Comment by vforgione** [comment]
Then I was right in my last comment..?

We can assume the environment is dev unless the value `MIX_ENV` is set to something else, which _is_ the case when testing and building. The build scripts can very easily set the environment name, and then shouldn't Webpack just be able to pick that up in `process.env.MIX_ENV`?

**Comment by vforgione** [comment]
Being merged with #422

**Comment by brlodi** [comment]
Sorry, I was wrong, this actually isn't part of #422, it'll be part of whatever PR closes #373.

I'll check it off in there and leave this closed, though.
